instrumentation/nocode/src/main/java/com/splunk/opentelemetry/instrumentation/nocode/JSPS.java

```diff
@@ -0,0 +1,173 @@
+/*
+ * Copyright Splunk Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.splunk.opentelemetry.instrumentation.nocode;
+
+import java.lang.reflect.Method;
+import java.lang.reflect.Modifier;
+import java.util.logging.Logger;
+
+/**
+ * JSPS stands for Java-like String-Producing Statement. A JSPS is essentially a single call in Java
+ * (as though it ends with a semicolon), with some limitations. Its purpose is to allow pieces of
+ * nocode instrumentation (attributes, span name) to be derived from the instrumentated context.
+ *
+ * <p>As some illustrative examples:
+ *
+ * <pre>
+ *   this.getHeaders().get("X-Custom-Header").substring(5)
+ *   param0.getDetails().getCustomerAccount().getAccountType()
+ * </pre>
+ *
+ * <p>The limitations are:
+ *
+ * <ul>
+ *   <li>no access to variables other than 'this' and 'paramN' (N indexed at 0)
+ *   <li>no control flow (if), no local variables, basically nothing other than a single chain of
+ *       method calls
+ *   <li>Method calls are limited to either 0 or 1 parameters currently
+ *   <li>Parameters must be literals and only integral (int/long), string, and boolean literals are
+ *       currently supported
+ * </ul>
+ */
+public final class JSPS {
```

Comment:
I think it is better to replace this with an existing expression language implementation as the parsing will only get more complicated as features are added. For example parsing `this.get(")")` will already be a challenge.